lib/cstruct.mli

```diff
@@ -214,13 +214,15 @@ val create_unsafe : int -> t
     to leak sensitive information.
 *)
 
-val of_string: ?allocator:(int -> t) -> string -> t
-(** [of_string ~allocator str] is the cstruct representation of [str],
+val of_string: ?allocator:(int -> t) -> ?off:int -> ?len:int -> string -> t
+(** [of_string ~allocator ~off ~len str] is the cstruct representation of [str]
+    slice located at [off] offset and of [len] length,
```

Comment:
Not directly relevant to this PR, but it may be worth explicitly clarifying that the memory is copied here, and there is no sharing between the fresh cstruct and the string or bytes.